dev_apps/uitest-privileged/js/API/settings.js

```diff
@@ -0,0 +1,47 @@
+var settingsTest = {
+  init: function () {
+    this.statusPlaceholder = document.getElementById('status-placeholder');
+    this.statusContainer = document.getElementById('status-container');
+
+    this.errorPlaceholder = document.getElementById('error-placeholder');
+    this.errorContainer = document.getElementById('error-container');
+  },
+  addStatus: function (aStatus) {
+    this.statusPlaceholder.innerHTML = this.statusPlaceholder.innerHTML + "<br\>" + aStatus;
+  },
+  onError: function(aError) {
+    console.error('ERROR ' + aError.name || aError);
+    this.errorContainer.classList.remove('hidden');
+    this.errorPlaceholder.textContent = aError.name;
+  },
+  runTest: function() {
+    if (navigator.mozSettings)
+    {    
+      var lock = navigator.mozSettings.createLock();
+      this.addStatus("Successfully created lock!");
+      var req = lock.get("wallpaper.image");
+      req.onsuccess = (() => {
+        this.addStatus("Successfully retrieved wallpaper");
+        var req2 = lock.set({"wallpaper.image": req.result["wallpaper.image"]});
+        req2.onsuccess = (() => {
+          this.addStatus("Successfully set wallpaper");
+        });
+        req2.onerror = (() => {
+          this.onError("COULD NOT SET WALLPAPER");
+        });
+      });
+      req.onerror = (() => {
+        this.onError("COULD NOT RETRIEVE WALLPAPER.")
+      });
+    }
+    else {
+      this.onError("SETTINGS API UNAVAILABLE");
```

Comment:
If I remove the permission from the manifest, then I see:
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App: Content JS ERROR: ERROR undefined 
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at settingsTest.onError (app://uitest-privileged.gaiamobile.org/js/API/settings.js:13:4)
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at settingsTest.runTest (app://uitest-privileged.gaiamobile.org/js/API/settings.js:38:6)
01-13 17:45:29.719  4056  4056 E UI tests - Privileged App:     at onload (app://uitest-privileged.gaiamobile.org/js/API/settings.js:46:2)

and the UI is not updated accordingly